mne/source_estimate.py

```diff
@@ -414,9 +420,12 @@ def guess_src_type():
     else:
         pass  # noqa
 
-    return Klass(
-        data=data, vertices=vertices, tmin=tmin, tstep=tstep, subject=subject
-    )
+    if is_kernel:
+        return Klass(data=(data, sens_data), vertices=vertices, tmin=tmin,
+                     tstep=tstep, subject=subject)
+    else:
+        return Klass(data=data, vertices=vertices, tmin=tmin, tstep=tstep,
+                     subject=subject)
```

Comment:
That's not how it works. 

There should be a single return.

Reply:
oh sry, I did it because of
https://github.com/mne-tools/mne-python/pull/6609#discussion_r309683694

But now I see what you mean. I'll change that

Reply:
f254a95cc7ed2b84db5107684045d2cc8fabda49 is not what I meant. I meant to keep a single 
`return Klass(..` to keep the logic we wrote in #6636. The entire purpose of #6636 was to keep this one simple.

Maybe I'm missing something. I'll give it a shot.

Reply:
Actually I already wrote it for you here https://github.com/mne-tools/mne-python/pull/6609#issuecomment-518783439

But I think that e41236f is even cleaner.

Reply:
> Actually I already wrote it for you here #6609 (comment)

Yes, it just seemed to me that you'd be most in favor of:
>```python
>+                stc = VectorSourceEstimate(
>+                    data=(data_rot, sens_data), vertices=vertices, tmin=tmin,
>+                    tstep=tstep, subject=subject
>+                )
>+            else:
>+                stc = VectorSourceEstimate(
>+                    data=data_rot, vertices=vertices, tmin=tmin,
>+                    tstep=tstep, subject=subject
>+                )
>```

But anyhow, I like your solution with the sentinel (tbh I didn't know about that practice). But it gets rid of that `is_kernel` bool and it's compact. So, for me, that's fine!

Reply:
The code is meant to read by people (machines executing it is secondary), and usually, the flow is: look at the definition, look at the return check whatever else afterward. If you have 2 returns or you construct the return in 2 different places (which is the same thing) I need to understand what's in the middle of the function. Most of the time when you read a function it is not for debugging purposes I just want to skim through it and get a good grasp of whats going on.

If I have a single place that returns `Klass` and does something at `data` and this is in the return, I need no further reading. This function is a factory. Based on whatever, chooses which class to construct and modifies the data accordingly.


@DiGyt this is a nice read on sentinel https://treyhunner.com/2019/03/unique-and-sentinel-values-in-python/